### PR TITLE
Replace enforce(!empty with assert(!empty at the beginning of range m…

### PR DESCRIPTION
**Pull request by nordlow**
…embers

### CODE REVIEW

File: std/base64.d [hunk 1]
```diff
@@ -639,7 +639,7 @@ template Base64Impl(char Map62th, char Map63th, char Padding = '=')
          */
         void popFront()
         {
-            enforce(!empty, new Base64Exception("Cannot call popFront on Encoder with no data remaining"));
+            assert(!empty, new Base64Exception("Cannot call popFront on Encoder with no data remaining"));
```
Comment (schveiguy) [inline std/base64.d:642]: I don't think you can do these.

Comment (pbackus) [reply std/base64.d:642]: We can get rid of the exception and just leave the string:
```suggestion
            assert(!empty, "Cannot call popFront on Encoder with no data remaining");
```

**Comment by maxhaton** [comment]
Does the rest of phobos use assertions or enforce?

If there was let's say, a large D project that rhymed with "Pill" that doesn't like assertions very much this an important consideration.

**Comment by pbackus** [comment]
`git grep '!empty'` shows an overwhelming majority for `assert`.

File: std/base64.d (hunk 1 above)
Comment (nordlow) [reply std/base64.d]: Adjusted.

**Review by dkorpel — APPROVED** [review]

**Comment by nordlow** [comment]
> > Does the rest of phobos use assertions or enforce?
> 
> Just on the std.container.array usage (and only in Array!bool), `enforce` was used when it was added 12 years ago. It looks like the original code was changed essentially to asserts: [27cc5e8](https://github.com/dlang/phobos/commit/27cc5e858f4f63903df2444d5206fcc2f1e8c441)
> 
> But the user forgot to do this to Array!bool.
> 
> So I think this is just a case of missed search/replace, or maybe the user had a specific issue they were addressing and weren't using Array!bool.

I'm satisfied. Ok to merge.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**